app.py

```diff
@@ -0,0 +1,168 @@
+    representation of entities in Polyglot as in spacy, the start and end index
+    relative to the actual text has to be calculated. Polyglot defines words and
+    uses word indices, while Spacy uses the char indices of the input string.
+    """
+    poly_text = Text(text, hint_language_code=lan)
+    ents = []
+    spans = []
+    entity_text = list(set([(' '.join(entity), entity.tag.lstrip('I-'))
+                            for entity in poly_text.entities]))
+    entity_text.sort(key=len, reverse=True)
+    for entity in entity_text:
+        for match in re.finditer(entity[0], text):
+            # the match is only a match if it's not inside another match.
+            # Therefore, the entities are ordered by length beforehand.
+            if all(not(span[0] <= match.start() <= span[1])
+                    and not(span[0] <= match.end() <= span[1]) for span in spans):
+                spans.append((match.start(), match.end()))
+                ents.append({'text': entity[0], 'start': match.start(),
+                            'end': match.end(), 'type': entity[1]})
+    return ents
+
+
+def get_spacy_ents(text, nlp):
+    """Processes an input text with a spacy model and returns a list of
+    entities.
+    The label for PERSON is changed in order to be in accordance with polyglot.
+    """
+    doc = nlp(text)
+    ents = [{'text': ent.text, 'start': ent.start_char,
+            'end': ent.end_char,
+             'label': 'PER' if ent.label_ == 'PERSON' else ent.label_}
+            for ent in doc.ents]
+    return ents
+
+
+def get_spacy_model(lan):
+    """Helper function for getting the most appropriate spacy model.
+    It assumes that the largest available model should be loaded and
+    the multilingual model only if no other model is available.
+    """
+    language_models = [model for model in DOWNLOADED_MODELS['spacy'] if model.startswith(lan)]
+    if not language_models:
+        method = 'xx_ent_wiki_sm'
+        nlp = spacy.load(method)
+        return method, nlp
+    for size in ['lg', 'md', 'sm']:
+        for model in language_models:
+            if model.endswith(size):
+                nlp = spacy.load(model)
+                return model, nlp
+
+
+def get_ents_from_model(text, lan, model):
+    """Fuction for checking what to do with a requested model.
+    For both, polyglot and spacy model first it is checked whether
+    they are downloaded. If not, the server with respond with a 500 error.
+    If they are downloaded, get the ents and return.
+    """
+    if model.startswith('poly'):
+        lan = model.split('_')[1]
+        if 'polyglot' in DOWNLOADED_MODELS and lan in DOWNLOADED_MODELS['polyglot']:
+            model = 'polyglot'
+            ents = get_poly_ents(text, lan)
+            return jsonify({'language': lan, 'model': model, 'entities': ents}), 200
+        else:
+            return jsonify({'error': f'no polyglot model for language {lan}'}), 500
```

Comment:
Sort out the errors first, so you can move the actual logic left a block:

```python
    if model.startswith('poly'):
        lan = model.split('_')[1]
        if 'polyglot' not in DOWNLOADED_MODELS or lan not in DOWNLOADED_MODELS['polyglot']:
            return jsonify({'error': f'no polyglot model for language {lan}'}), 500
        model = 'polyglot'
        ents = get_poly_ents(text, lan)
        return jsonify({'language': lan, 'model': model, 'entities': ents}), 200
    else: ...
```

Reply:
I actually don't understand how ^^
If the model name starts with poly, I have to check whether the language is part of the downloaded polyglot models.
If not, I have to check whether the model name is part of the downloaded spacy models.

Reply:
I didn't say to remove all the ifs, just the inner ones that should treat the errors first.